test/rr-tests.js

```diff
@@ -0,0 +1,286 @@
+      spec: {
+        clusterAuth: {
+          impersonateUser: 'razeedeploy'
+        },
+        backendService: 'git',
+        requests: [
+          {
+            options: {
+              git: {
+                provider: 'github',
+                repo: 'https://github.com/razee-io/RemoteResource.git',
+                ref: 'tests',
+                filePath: 'test/test-configs/test-config.yaml'
+              }
+            }
+          }
+        ]
+      }
+        
+    }
+  };
+
+  const eventDataGit1 = {
+    object: {
+      apiVersion: 'deploy.razee.io/v1alpha2',
+      kind: 'RemoteResource',
+      metadata: {
+        name: 'rr',
+        namespace: 'default'
+      },
+      spec: {
+        clusterAuth: {
+          impersonateUser: 'razeedeploy'
+        },
+        backendService: 'git',
+        requests: [
+          {
+            options: {
+              git: {
+                provider: 'github',
+                repo: 'https://github.com/razee-io/RemoteResource.git',
+                ref: 'tests',
+                filePath: 'test/test-configs/*.yaml'
+              }
+            }
+          }
+        ]
+      }
+        
+    }
+  };
+
+  const eventDataS3 = {
+    object: {
+      apiVersion: 'deploy.razee.io/v1alpha2',
+      kind: 'RemoteResource',
+      metadata: {
+        name: 'rr',
+        namespace: 'default'
+      },
+      spec: {
+        clusterAuth: {
+          impersonateUser: 'razeedeploy'
+        },
+        backendService: 's3',
+        iam: {
+          url: 'https://iam.cloud.ibm.com/identity/token',
+          grantType: 'urn:ibm:params:oauth:grant-type:apikey',
+          apiKey: 'testApiKey'
+        },
+        requests: [
+          {
+            options: {
+              url: 'https://s3.us.cloud-object-storage.appdomain.cloud/bucket/test-config.yaml'
+            }
+          }
+        ]
+      }
+        
+    }
+  };
+
+  const eventDataS3_1 = {
+    object: {
+      apiVersion: 'deploy.razee.io/v1alpha2',
+      kind: 'RemoteResource',
+      metadata: {
+        name: 'rr',
+        namespace: 'default'
+      },
+      spec: {
+        clusterAuth: {
+          impersonateUser: 'razeedeploy'
+        },
+        backendService: 's3',
+        iam: {
+          url: 'https://iam.cloud.ibm.com/identity/token',
+          grantType: 'urn:ibm:params:oauth:grant-type:apikey',
+          apiKey: 'testApiKey'
+        },
+        requests: [
+          {
+            options: {
+              url: 'https://s3.us.cloud-object-storage.appdomain.cloud/bucket/'
+            }
+          }
+        ]
+      }
+        
+    }
+  };
+
+  it('Create generic RemoteResourceController', async function () {
+    // backendServiceFactory should create RemoteResourceController for generic backendService
+    const controller = setupController(eventData);
+    assert(controller instanceof RemoteResourceController);
+  });
+
+  it('RRGitController single file request', async function() {
```

Comment:
Mock the `request` so the test doesn't depend on how the git repo is setup.

Reply:
already mocked with getFilesStub